tests/js/couchConfigTests.js

```diff
@@ -0,0 +1,219 @@
+                args: ["Database create/verify was completed successfully"]
+            }]
+        },
+        {
+            name: "Test CouchDB document loading",
+            expect: 5,
+            sequence: [{
+                "func": "{couchConfigTest}.couchConfig.ensureDBExists"
+            },
+            {
+                "event": "{couchConfig}.events.onDBExists",
+                "listener": "{couchConfigTest}.couchConfig.updateDocuments"
+            },
+            {
+                "func": "jqUnit.assertEquals",
+                "args": ["Total documents count is expected number", 2, "{couchConfigTest}.couchConfig.totalDocuments"]
+            },
+            {
+                "event": "{couchConfig}.events.onDocsUpdated",
+                "listener": "jqUnit.assert",
+                args: ["Database documents were created/updated successfully"]
+            },
+            {
+                "func": "fluid.tests.couchConfig.testDbDocument",
+                args: ["{couchConfigTest}.couchConfig.options.dbConfig.dbName",
+                    "{couchConfigTest}.couchConfig.options.dbConfig.couchURL",
+                    "{couchConfigTest}.couchConfig.options.dbDocuments.testDoc",
+                    "{that}.events.nanoCallBackDone"]
+            },
+            {
+                "event": "{that}.events.nanoCallBackDone",
+                "listener": "jqUnit.assert",
+                args: ["End of test sequence"]
+            }]
+        },
+        {
+            name: "Test CouchDB design document loading",
+            expect: 5,
+            sequence: [{
+                "func": "{couchConfigTest}.couchConfig.ensureDBExists"
+            },
+            {
+                "event": "{couchConfig}.events.onDBExists",
+                "listener": "{couchConfigTest}.couchConfig.updateDesignDoc"
+            },
+            {
+                "event": "{couchConfig}.events.onDesignDocUpdated",
+                "listener": "jqUnit.assert",
+                args: ["Database design document was created/updated successfully"]
+            },
+            {
+                "func": "fluid.tests.couchConfig.testDbView",
+                args: ["{couchConfigTest}.couchConfig.options.dbConfig.dbName",
+                    "{couchConfigTest}.couchConfig.options.dbConfig.couchURL",
+                    "{couchConfigTest}.couchConfig.options.dbViews.test",
+                    "{couchConfigTest}.couchConfig.options.dbValidate.validateFunction",
+                    "{that}.events.nanoCallBackDone"]
+            },
+            {
+                "event": "{that}.events.nanoCallBackDone",
+                "listener": "jqUnit.assert",
+                args: ["End of test sequence"]
+            }]
+        }]
+    }]
+});
+
+// A basic validation that checks the document to make sure its 'type' is 'test'
+// eslint-disable-next-line no-unused-vars
+fluid.tests.couchConfig.testValidateFunction = function (newDoc, oldDoc, userCtx) {
```

Comment:
I think that we are using "test" in 2 different ways in our naming of functions here.

We are using it as a descriptor (as in, a function used in our testing):

- testValidateFunction
- testMapFunction
- testReduceFunction

And we are using it as a verb (as in, do the tests for this document):

- testDbDocument
- testDbView

I think that the first case is probably most consistent with our common usage: a test, a test case, a test environment. And maybe for the verb case, we could use "assert", "verify", or "checkThat"?